lib/rspec_controller_helpers/resourceful.rb

```diff
@@ -1,23 +1,43 @@
 RSpec.shared_examples_for 'a resourceful controller' do |model, raw_options|
-  let(:options) { raw_options || {} }
+  let(:options) { ({ endpoint_formats: {} }).merge(raw_options) }
   let(:symbol) { options[:symbol] || model.name.downcase.to_sym }
   let(:factory) { options[:factory] || symbol }
 
-  let(:tests) do
+  let(:endpoints_to_test) do
     all = [:index, :new, :create, :show, :edit, :update, :delete]
     except = options[:except] || []
     options[:only] || (all - except)
   end
 
+  let(:endpoint_formats) do
+    endpoints_to_test
+      .map{|t| { t => :html } }
+      .reduce(:merge)
+      .merge(options[:endpoint_formats])
+  end
+
+  let(:json_response) { JSON.parse(response.body) }
+
   describe "GET /<resource>" do
     let!(:objects) { create_list(factory, 3) }
     subject { get :index }
 
-    it "renders the index page" do
-      if tests.include? :index
+    it 'is successful' do
+      if endpoints_to_test.include? :index
         subject
         expect(response).to have_http_status(:ok)
-        expect(response).to render_template(:index)
+      end
+    end
+
+    it "renders the index page" do
+      if endpoints_to_test.include? :index
+        subject
+        case endpoint_formats[:index]
+        when :html
+          expect(response).to render_template(:index)
+        when :json
+          expect(json_response.length).to eq(3)
+        end
```

Comment:
So we can only test one or the other? What about making endpoint_formats an array to each endpoint/key?
```
formats[:index].each do |format|
  if format == :html
   <blah></blah>
  elsif format == :json
   { "blah": "blah"} 
  elsif format == :xml (future PR that will be ridiculed)
    puts "ewww"
  end
end
```